test/gtest/common/mem_buffer.h

```diff
@@ -67,6 +67,9 @@ class mem_buffer {
     /* returns whether any other type of memory besides the CPU is supported */
     static bool is_gpu_supported();
 
+    /* set device context if compiled with GPU support */
+    static void set_device_context();
```

Comment:
Is there an example of STATIC_INIT in gtest?

Reply:
in C++: 
option1: declare a global variable (static class member) whose value is initialized as the result of some function which actually sets cuda device
option2: create an internal class whose constructor sets cuda device and declare a global variable of that class type
option3: in every call to mem buffer allocate, check som global variable if cuda device was already set, if not - set it

Reply:
@yosefe Made changes to call setdevice (if not already set per thread) only before calling uct_mem_alloc. Setting device context in mem buffer allocate doesn't suffice because we need to set context before we call UCT API with cuda_cpy md and eventually call cuMemAlloc.

Reply:
@yosefe 

Looks like I'm seeing compilation issues with use of `thread_local`. Is it because C++ 11 isn't supported by the compiler? 
```
/scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/mem_buffer.cc: In static member function ‘static void mem_buffer::set_device_context()’:
/scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/mem_buffer.cc:95:5: error: ‘thread_local’ was not declared in this scope
     thread_local bool device_not_set = true;
     ^
/scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/mem_buffer.cc:95:18: error: expected ‘;’ before ‘bool’
     thread_local bool device_not_set = true;
                  ^
/scrap/azure/agent-03/AZP_WORKSPACE/2/s/contrib/../test/gtest/common/mem_buffer.cc:97:9: error: ‘device_not_set’ was not declared in this scope
     if (device_not_set) {
         ^
```

Reply:
static __thread bool device_not_set = true; ?